terraform_compliance/extensions/terraform.py

```diff
@@ -573,3 +571,36 @@ def is_type(self, resource, mode):
             return resource['address'].split('.')[0] == mode
 
         return False
+
+    def process_module_calls(self, module_resource, parents_modules=None):
+        if parents_modules is None:
+            parents_modules = []
+
+        resources = []
+        for k, v in module_resource.items():
+            # Set the naming correct (for cases like module.a.module.b.module.c...)
+            current_module_level = parents_modules
+            current_module_level.append('module.{}'.format(k))
+            module_name = ".".join(current_module_level)
+
+            # Register the resource (along with module naming)
+            if 'resources' in v.get('module', {}):
+                for resource in v['module']['resources']:
+                    resource['address'] = '{}.{}'.format(module_name, resource['address'])
+                    resources.append(resource)
+
+            # Dive deeper, its not finished yet.
+            if 'module_calls' in v.get('module', {}):
+                resources.extend(self.process_module_calls(v['module']['module_calls'], current_module_level))
+
+        return resources
+
+    def extract_resource_type_from_address(self, resource_address_string):
+        if '.' in resource_address_string:
+            octets = resource_address_string.split('.')
+            if len(octets) > 1:
+                return octets[-2]
+            else:
+                return octets[0]
+        return resource_address_string
```

Comment:
Could you add some comments to this function, explaining the reasoning of how address is extracted? An example of how intended cases should work would be great as well. It will be easier to go back to this if we have a problem with module naming in the future.

Reply:
Good spot, added on 1c3f408